src/main/java/org/threadly/litesockets/networkutils/ProfileServer.java

```diff
@@ -0,0 +1,213 @@
+    HELP = ByteBuffer.wrap(sb.toString().getBytes()).asReadOnlyBuffer();
+  }
+
+  private final SimpleSchedulerInterface scheduler;
+  private final SocketExecuterInterface socketEx;
+  private final ConcurrentHashMap<Client, MergedByteBuffers> clients = new ConcurrentHashMap<Client, MergedByteBuffers>();
+  private final Profiler profiler;
+  private final String host;
+  private final int port;
+  private TCPServer server;
+  
+  
+  /**
+   * Construct a ProileServer
+   * 
+   * @param socketEx The SocketExecuterInterface to use with this Server.
+   * @param host the host to create the servers listen port on.
+   * @param port the port to use.
+   * @param frequency the frequency of the profiler in ms. 
+   */
+  public ProfileServer(SocketExecuterInterface socketEx, String host, int port, int frequency) {
+    scheduler = socketEx.getThreadScheduler();
+    this.socketEx = socketEx;
+    profiler = new Profiler(frequency);
+    this.host = host;
+    this.port = port;
+  }
+  
+  @Override
+  public void onClose(Client client) {
+    synchronized(clients) {
+      clients.remove(client);
+    }
+  }
+
+  @Override
+  public void onRead(final Client client) {
+    MergedByteBuffers mbb = clients.get(client);
+    mbb.add(client.getRead());
+    int pos = -1;
+    while((pos = mbb.indexOf("\n")) > -1) {
+      String cmd = mbb.getAsString(pos).trim().toLowerCase();
+      mbb.discard(1);
+      if(START_PROFILE.equals(cmd)) {
+        if(!profiler.isRunning()) {
+          profiler.start();
+          client.writeForce(STARTED_RESPONSE.duplicate());
+        } else {
+          client.writeForce(ALREADY_STARTED_RESPONSE.duplicate());
+        }
+      } else if(STOP_PROFILE.equals(cmd)) {
+        if(profiler.isRunning()) {
+          profiler.stop();
+          client.writeForce(STOPPED_RESPONSE.duplicate());
+        } else {
+          client.writeForce(ALREADY_STOPPED_RESPONSE.duplicate());
+        }
+      } else if(RESET_PROFILE.equals(cmd)) {
+        profiler.reset();
+        client.writeForce(RESET_RESPONSE.duplicate());
+      } else if(DUMP_PROFILE.equals(cmd)) {
+        dumpProfile(client);
+      } else {
+        sendHelp(client);
+      }
+    }
+    if(mbb.remaining() > 100) {
+      client.setReader(null);
+      client.writeForce(BAD_DATA.duplicate());
+      this.scheduler.schedule(new Runnable() {
+        @Override
+        public void run() {
+          client.close();
+        }}, 500);
+    }
+        
+  }
+
+  @Override
+  public void accept(Client client){
+    try {
+      clients.put(client, new MergedByteBuffers());
+      client.setReader(this);
+      client.setCloser(this);
+      socketEx.addClient(client);
+    } catch (Exception e) {
+      ExceptionUtils.handleException(e);
+    }
+  }
+  
+  private void sendHelp(final Client client) {
+    scheduler.execute(new Runnable() {
+      @Override
+      public void run() {
+        client.writeForce(HELP.duplicate());
```

Comment:
since its a writeForce the write is immediately put on the Client Objects buffer, and then written out as it can.  So this call will complete as fast as the BB can be added to the ArrayDeque in the client.

If the buffer was not fully sent out to the client by the time we got a close, the client object would be removed from the client map and be cleaned up with any remaining data in the buffer.

If it was closed right before we got here this most likely be a no-op, worst case the close happens right after the write check, at which point the client objects write buffer has that data, but once the function ends the client object should be orphaned for GC.